### ISSUE TITLE
NewRadioGroup panics: invalid memory address or nil pointer dereference

### ISSUE THREAD

**Issue by amo13** [fyne-io/fyne#2346]
### Describe the bug:
NewRadioGroup does not work and produces the following output:
```
2021/08/02 16:09:37 Fyne error:  Attempt to access current Fyne app when none is started
2021/08/02 16:09:37   At: /home/amo/go/pkg/mod/fyne.io/fyne/v2@v2.0.3/app.go:79
2021/08/02 16:09:37 Fyne error:  Attempt to access current Fyne app when none is started
2021/08/02 16:09:37   At: /home/amo/go/pkg/mod/fyne.io/fyne/v2@v2.0.3/app.go:79
2021/08/02 16:09:37 Fyne error:  Attempt to access current Fyne app when none is started
2021/08/02 16:09:37   At: /home/amo/go/pkg/mod/fyne.io/fyne/v2@v2.0.3/app.go:79
2021/08/02 16:09:37 Fyne error:  Attempt to access current Fyne app when none is started
2021/08/02 16:09:37   At: /home/amo/go/pkg/mod/fyne.io/fyne/v2@v2.0.3/app.go:79
2021/08/02 16:09:37 Fyne error:  Attempt to access current Fyne app when none is started
2021/08/02 16:09:37   At: /home/amo/go/pkg/mod/fyne.io/fyne/v2@v2.0.3/app.go:79
2021/08/02 16:09:37 Fyne error:  Attempt to access current Fyne app when none is started
2021/08/02 16:09:37   At: /home/amo/go/pkg/mod/fyne.io/fyne/v2@v2.0.3/app.go:79
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x60 pc=0x5c494f]

goroutine 1 [running, locked to thread]:
fyne.io/fyne/v2/canvas.NewText(0x7c26e7, 0x11, 0x823ea0, 0xc000016d20, 0x10)
	/home/amo/go/pkg/mod/fyne.io/fyne/v2@v2.0.3/canvas/text.go:46 +0x2f
fyne.io/fyne/v2/widget.(*radioItem).CreateRenderer(0xc0002ba230, 0x7aadc0, 0xc0002ba230)
	/home/amo/go/pkg/mod/fyne.io/fyne/v2@v2.0.3/widget/radio_item.go:40 +0x10d
fyne.io/fyne/v2/internal/cache.Renderer(0x82b380, 0xc0002ba230, 0xc0002ba230, 0x7de0b0)
	/home/amo/go/pkg/mod/fyne.io/fyne/v2@v2.0.3/internal/cache/widget.go:29 +0x119
fyne.io/fyne/v2/widget.(*BaseWidget).Refresh(0xc0002ba230)
	/home/amo/go/pkg/mod/fyne.io/fyne/v2@v2.0.3/widget/widget.go:138 +0x4b
fyne.io/fyne/v2/widget.(*RadioGroup).update(0xc00011d9a0)
	/home/amo/go/pkg/mod/fyne.io/fyne/v2@v2.0.3/widget/radio_group.go:129 +0x2db
fyne.io/fyne/v2/widget.NewRadioGroup(0xc000062420, 0x2, 0x2, 0x7dda78, 0x7f84db52c108)
	/home/amo/go/pkg/mod/fyne.io/fyne/v2@v2.0.3/widget/radio_group.go:36 +0x93
main.init()
	/home/amo/go/test/test.go:9 +0x85
exit status 2
```

### To Reproduce / Example Code
```go
package main

import(
	"fyne.io/fyne/v2/widget"
	"fyne.io/fyne/v2/app"
)

var Radio_rom_source = widget.NewRadioGroup([]string{"Official Releases", "Archive"}, changedRomSource)

func changedRomSource(source string) {

}

func main() {
	a := app.New()

	w := a.NewWindow("Testing")
	w.SetMaster()

	w.SetContent(Radio_rom_source)

	w.ShowAndRun()
}
```
Change `NewRadioGroup` to `NewSelect` and no error is thrown.
According to [the fyne tour](https://developer.fyne.io/tour/widget/choices.html), they should be no other code-wise differences between Select and RadioGroup.

### Device (please complete the following information):
 - **OS:** Linux
 - **Version:** 5.10.42-1-MANJARO
 - **Go version:**go1.16.6 linux/amd64
 - **Fyne version:** 2.0.3

**Comment by amo13** [comment]
Oh wow, I think after a few more of these on different objects, I understand the problem:
I have to declare the variable this way: `var Radio_rom_source *widget.RadioGroup` outside a function and define it later on inside a function with `Radio_rom_source = widget.NewRadioGroup([]string{"Official Releases", "Archive"}, changedRomSource)`.

Not sure if this is expected, especially since it *does* work with Select instead of RadioGroup...

Feel free to close this since I found a workaround, but I'd appreciate any hint on why this is happening and/or why I am or am not supposed to do that. (I am still very new to go, so I might not know.)

**Comment by Jacalz** [comment]
I really don’t think there is something here that we can do about it. The panic is, like the error above mentions, because your widget is created before the application has been started. Further yet, global variables is, in many scenarios, often a bad idiom I’m coding (apart for things like iota, constants and such). See https://www.tutorialspoint.com/Why-should-we-avoid-using-global-variables-in-C-Cplusplus. It is also the case that it is a strain on the Go GC because the global variables can never be freed (might not matter much in this simple example when the widget is used forever, but is more important in larger programs) and thus always take up memory.

**Comment by amo13** [comment]
I understand. Thank you a lot for your answer, I appreciate that!

I guess that my use of global variables relies on my lack of better ideas. I need to access some variables outside the scope of the declaring functions and sometimes also outside the scope of my gui package. Do you have a hint on how to achieve that in a more idiomatic way?

And just out of curiosity: Why is creating a Select widget before the application is started okay and creating a RadioGroup widget before the application is started not okay?

**Comment by andydotxyz** [comment]
We should probably fix the crash, as Fyne should never crash, however this is (as the warnings point out) a bad thing.
You can define the variable globally if you have to, but initialise it later, after the app is created.

**Comment by Jacalz** [comment]
I think that we most likely are going to find that a lot of widgets may or may not panic when used as global variables.

**Comment by andydotxyz** [comment]
The panic is fixed on `release/v2.0.x` branch ready for v2.0.4.
